Price section - Small refactor and style fix

octaedro opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->
This PR fixes a few things related to the prices section and addresses a couple of suggestions made for [this PR](https://github.com/woocommerce/woocommerce/pull/34382).

It adds:
- a small visual bugfix (the help tooltip popped up too high)
- a refactor of the method to sanitize the price.
- a change in the Sale price validation (now the Sale price has to be equal to or higher than the List price).
- an error message when the `Sale price` has an error.

Closes # 38-gh-woocommerce/mothra-private.

### How to test the changes in this Pull Request:

1. Verify that the `Feature this product` tooltip looks and works correctly.

![Screen Shot 2022-09-07 at 12 26 57](https://user-images.githubusercontent.com/1314156/189352458-020bfbdd-a2a0-49bd-ba40-fe2aa79e3dd9.png)

2. Verify that the prices look correct. Check the testing [instructions here](https://github.com/woocommerce/woocommerce/pull/34382) too.
3. Create a new product without `List price` but with a `Sale price` and verify that the error message `Sale price cannot be equal to or higher than list price.` is being shown after the `Sale price` loses the focus.
4. Set a `Sale price` equal to or higher than the `List price` and verify that the warning `Sale price cannot be equal to or higher than list price.` is visible under the `Sale price`.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

octaedro committed:
Fix modal

octaedro committed:
Add method sanitizePrice to `useProductHelper`

# Conflicts:
#	plugins/woocommerce-admin/client/products/sections/pricing-section.tsx

# Conflicts:
#	plugins/woocommerce-admin/client/products/sections/pricing-section.tsx

octaedro committed:
Add changelog

octaedro committed:
Add sales validation

octaedro committed:
Add sale price validation

octaedro committed:
Fix regularPriceProps

octaedro committed:
Set list price when it's empty

octaedro committed:
Fix Sale price validation

louwie17 commented on plugins/woocommerce-admin/client/products/product-form-actions.tsx:
```diff
@@ -42,12 +42,23 @@ export const ProductFormActions: React.FC = () => {
 
+	const maybeSetRegularPrice = ( product: Product ) => {
+		if (
+			product.regular_price === undefined &&
```
Should we also check if `regular_price` is an empty string?
As this will break as soon as the user edits the list price, but clears it.

octaedro replied:
Nice catch! This would be redundant considering Maikel's [PR here](https://github.com/woocommerce/woocommerce/pull/34658). I removed the method `maybeSetRegularPrice` in commit [0e65863](https://github.com/woocommerce/woocommerce/pull/34558/commits/0e6586341082a7e33a22a24602fa9b4e89fe39d9)

louwie17 commented on plugins/woocommerce-admin/client/products/use-product-helper.ts:
```diff
@@ -48,6 +57,9 @@ export function useProductHelper() {
 	} );
+	const context = useContext( CurrencyContext );
+	const { getCurrencyConfig } = context;
+	const { decimalSeparator } = getCurrencyConfig();
```
Should we move the above two lines into the `sanitizePrice` function? given the `decimealSeperator` isn't used anywhere else, and would needlessly be triggered on every re-render. It's probably not super JS extensive, but it be nice to keep that stuff to a minimum.

octaedro replied:
Good idea! I changed it in commit [b60ef79](https://github.com/woocommerce/woocommerce/pull/34558/commits/b60ef7953cb9944bed74553e8196cbe98bffb498)

louwie17 reviewed (commented):
Overall nice work @octaedro, I just left a couple comments and one in relation to this test instruction:

> Create a new product without List price but with a Sale price and verify that the List price is set and the Sale price has been deleted after saving. This should happen after pressing Save draft, or Publish, or Publish & duplicate, or Copy to a new draft.

This only worked for me when I never touched the list price and only added the sale price. But as soon as I edited the list price and cleared it, this logic wouldn't work anymore. I believe this would be the same for existing products.
I left a comment within the code to check for empty string as well.

Also I was curious if this logic would be semi redundant once @mdperez86 PR is in with disabling the action buttons?

octaedro committed:
Check empty regular_price

octaedro committed:
Small refactor to use-product-helper

octaedro committed:
Remove method `maybeSetRegularPrice`

octaedro commented:
Thank you @louwie17 for your review,
I just updated the testing instructions (they were outdated) and addressed the suggestions you made.
Could you take another look at this PR?

louwie17 reviewed (approved):
Thanks for the updates on this @octaedro, this tested well and changes look good :)
